### PR TITLE
Pub/Sub IAM

### PR DESCRIPTION
**Pull request by blowmage**
The Pub/Sub IAM Policy JSON structure has changed. (See #395). Update the code to use the latest Pub/Sub JSON and behavior. Add acceptance tests for getting and setting the policy.

### CODE REVIEW

**Comment by jeffmendoza** [comment]
+1

**Comment by quartzmo** [comment]
Thanks again @jeffmendoza for surfacing the IAM change! Apparently there is no versioning on the IAM methods in the REST API(s)?

**Comment by blowmage** [comment]
I have acceptance tests for the `testIamPermissions` endpoint, but they are failing right now with the following:

```yaml
error:
  code: 503
  message: The service was unable to fulfill your request. Please try again.
  errors:
  - message: The service was unable to fulfill your request. Please try again.
    domain: global
    reason: backendError
  status: UNAVAILABLE
```

AFAICT this doesn't look to be a problem with how the ruby code is calling the endpoint. Do we want to hold off on the acceptance tests for now? Or should I add `skip` to them until they start to pass?

**Comment by quartzmo** [comment]
If you already have them, keeping them with `skip` sounds ok to me. We just need to remember to try them again periodically!

File: lib/gcloud/pubsub/subscription.rb [hunk 1]
```diff
@@ -476,11 +476,11 @@ def delay new_deadline, *messages
       # A hash that conforms to the following structure:
       #
       #   {
+      #     "etag"=>"CAE=",
```
Comment (quartzmo) [inline lib/gcloud/pubsub/subscription.rb]: What about the `version` attribute?

Comment (blowmage) [inline lib/gcloud/pubsub/subscription.rb]: The `version` attribute is in the documentation, but not in the data being returned by the service. I decided to trust the service's behavior over the documentation so I left `version` out.

Comment (quartzmo) [inline lib/gcloud/pubsub/subscription.rb]: Oh! OK, good call.

**Comment by quartzmo** [comment]
This looks good to me. @blowmage, can I merge?

**Comment by blowmage** [comment]
Yep. Merge away. :shipit:

**Pull request merged**